kv/kvserver: TestReplicaCircuitBreaker_Follower_QuorumLoss failed

kv/kvserver.TestReplicaCircuitBreaker_Follower_QuorumLoss [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21132430?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21132430?buildTab=artifacts#/) on master @ [a81bc2a9b579390be4abf6c9920ef21ff8d9e506](https://github.com/cockroachdb/cockroach/commits/a81bc2a9b579390be4abf6c9920ef21ff8d9e506):


```
             0.084ms      0.010ms    event:kv/kvserver/concurrency/concurrency_manager.go:426 [n1,s1,r82/1:/{Table/Max-Max}] scanning lock table for conflicting locks
             0.088ms      0.004ms        === operation:executeWriteBatch gid:31548 _verbose:1 node:1 store:1 range:82/1:/{Table/Max-Max}
             0.142ms      0.058ms    event:kv/kvserver/replica_send.go:530 [n1,s1,r82/1:/{Table/Max-Max}] concurrency retry error: invalid lease
             0.305ms      0.163ms    event:kv/kvserver/replica_send.go:204 [n1,s1,r82/1:/{Table/Max-Max}] replica.Send got error: replica unavailable: (n1,s1):1 unable to serve request to r82:/{Table/Max-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]: closed timestamp: 1771539229.911163197,0 (2026-02-19 22:13:49); raft status: {"id":"1","term":7,"vote":"2","commit":16,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":16,"progress":{},"leadtransferee":"0"}: replica has been leaderless for 1m0s
    soon.go:71: saw unexpected error replica unavailable: (n1,s1):1 unable to serve request to r82:/{Table/Max-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]: closed timestamp: 1771539229.911163197,0 (2026-02-19 22:13:49); raft status: {"id":"1","term":7,"vote":"2","commit":16,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":16,"progress":{},"leadtransferee":"0"}: replica has been leaderless for 1m0s
        (1) replica unavailable: (n1,s1):1 unable to serve request to r82:/{Table/Max-Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3, sticky=9223372036.854775807,2147483647]: closed timestamp: 1771539229.911163197,0 (2026-02-19 22:13:49); raft status: {"id":"1","term":7,"vote":"2","commit":16,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":16,"progress":{},"leadtransferee":"0"}: replica has been leaderless for 1m0s
        Wraps: (2)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.replicaUnavailableError
          | 	pkg/kv/kvserver/replica_circuit_breaker.go:272
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).replicaUnavailableErrorRLocked
          | 	pkg/kv/kvserver/replica_circuit_breaker.go:295
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*leaderlessWatcher).refreshUnavailableState
          | 	pkg/kv/kvserver/replica.go:302
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).tick
          | 	pkg/kv/kvserver/replica_raft.go:1515
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processTick
          | 	pkg/kv/kvserver/store_raft.go:789
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
          | 	pkg/kv/kvserver/scheduler.go:412
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
          | 	pkg/kv/kvserver/scheduler.go:314
          | runtime.goexit
          | 	src/runtime/asm_arm64.s:1268
        Wraps: (3) closed timestamp: 1771539229.911163197,0 (2026-02-19 22:13:49); raft status: {"id":"1","term":7,"vote":"2","commit":16,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":16,"progress":{},"leadtransferee":"0"}
        Wraps: (4)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*leaderlessWatcher).refreshUnavailableState
          | 	pkg/kv/kvserver/replica.go:303
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).tick
          | 	pkg/kv/kvserver/replica_raft.go:1515
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processTick
          | 	pkg/kv/kvserver/store_raft.go:789
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
          | 	pkg/kv/kvserver/scheduler.go:412
          | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
          | 	pkg/kv/kvserver/scheduler.go:314
          | runtime.goexit
          | 	src/runtime/asm_arm64.s:1268
        Wraps: (5) replica has been leaderless for 1m0s
        Error types: (1) *kvpb.ReplicaUnavailableError (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.leafError
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestReplicaCircuitBreaker_Follower_QuorumLoss4075261955
--- FAIL: TestReplicaCircuitBreaker_Follower_QuorumLoss (62.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaCircuitBreaker_Follower_QuorumLoss.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60573

--- a/pkg/kv/kvserver/client_replica_circuit_breaker_test.go
+++ b/pkg/kv/kvserver/client_replica_circuit_breaker_test.go
@@ -1063,6 +1063,10 @@ func setupCircuitBreakerTest(t *testing.T, leaseType roachpb.LeaseType) *circuit
 		kvserver.OverrideDefaultLeaseType(ctx, &st.SV, leaseType)
 	}
 
+	// Disable the leaderless watcher; it is tested separately and can
+	// interfere with circuit breaker assertions. See #163999.
+	kvserver.ReplicaLeaderlessUnavailableThreshold.Override(ctx, &st.SV, 0)
+
 	storeKnobs := &kvserver.StoreTestingKnobs{
 		SlowReplicationThresholdOverride: func(ba *kvpb.BatchRequest) time.Duration {
 			t.Helper()

